### ISSUE TITLE
[Bug]: when topK is set 300, parsing search results is time-consuming

### ISSUE THREAD

**Issue by zk903956028** [milvus-io/milvus#26079]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When the topk is set to a large value, parsing search results using the Java API is slow

### Expected Behavior

Search result parsing can be completed in milliseconds just like using the ATTU interface

### Steps To Reproduce

```markdown
no
```

### Milvus Log

no

### Anything else?

no

**Comment by xiaofan-luan** [comment]
could you offer detailed profiling result?
when you are sasying parse result , what is the most time consumption function?

**Comment by zk903956028** [comment]
which milvus version is V2.2.11,and the java sdk version is v2.2.9

**Comment by zk903956028** [comment]
> could you offer detailed profiling result? when you are sasying parse result , what is the most time consumption function?



> could you offer detailed profiling result? when you are sasying parse result , what is the most time consumption function?
R<SearchResults> respSearch = milvusClient.search(builder.build());
SearchResultsWrapper wrapperSearch = new SearchResultsWrapper(respSearch.getData().getResults());
List<QueryResultsWrapper.RowRecord> rowRecords = wrapperSearch.getRowRecords();
List<Map<String, Object>> data = new ArrayList<>();
rowRecords.stream().forEach(item -> data.add(item.getFieldValues()));

The above is a code example, and the time-consuming sentence is List<QueryResultsWrapper. RowRecord>rowRecords=wrapperSearch. getRowRecords()

**Comment by yanliang567** [comment]
/assign @yongpengli-z  @yhmo 
any ideas?

**Comment by yhmo** [comment]
@zk903956028 
The results returned by search() are column-based.
The SearchResultsWrapper decodes the column-based results to be row-based rows. With topk=10000, the time cost for decoding is 1.x second. The decoding method is a circle:
```
            for (int n = 0; n < 10000; ++n) {
                idScores.add(new IDScore("", longIDs.getData((int)offset + n), results.getScores((int)offset + n)));
            }
```

If you don't need the row-based results, you can directly parse the SearchResults struct returned by search().

**Comment by xiaofan-luan** [comment]
> @zk903956028 The results returned by search() are column-based. The SearchResultsWrapper decodes the column-based results to be row-based rows. With topk=10000, the time cost for decoding is 1.x second. The decoding method is a circle:
> 
> ```
>             for (int n = 0; n < 10000; ++n) {
>                 idScores.add(new IDScore("", longIDs.getData((int)offset + n), results.getScores((int)offset + n)));
>             }
> ```
> 
> I didn't see a good solution to improve this performance. If you don't need the row-based results, you can directly parse the SearchResults struct returned by search().

maybe  longIDs.getData((int)offset + n) is slow?
should be just 

> 

how long does it take for topk = 300?

**Comment by zk903956028** [comment]
> > @zk903956028 The results returned by search() are column-based. The SearchResultsWrapper decodes the column-based results to be row-based rows. With topk=10000, the time cost for decoding is 1.x second. The decoding method is a circle:
> > ```
> >             for (int n = 0; n < 10000; ++n) {
> >                 idScores.add(new IDScore("", longIDs.getData((int)offset + n), results.getScores((int)offset + n)));
> >             }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I didn't see a good solution to improve this performance. If you don't need the row-based results, you can directly parse the SearchResults struct returned by search().
> 
> maybe longIDs.getData((int)offset + n) is slow? should be just
> 
> > 
> 
> how long does it take for topk = 300?

when the topk =300, List<QueryResultsWrapper.RowRecord> rowRecords = wrapperSearch.getRowRecords(); 
this line cost about 10.x second, and need the row-based results!

**Comment by zk903956028** [comment]
System.out.println("Search results:");
for (int i = 0; i < targetVectors.size(); ++i) {
    List<SearchResultsWrapper.IDScore> scores = results.getIDScore(i);
    for (SearchResultsWrapper.IDScore score:scores) {
        System.out.println(score);
    }
}
when i use above, which from 'https://milvus.io/api-reference/java/v2.2.x', 
the problem has been resolved

**Comment by xiaofan-luan** [comment]
mark it as good first issue

**Comment by yhmo** [comment]
A performance issue of SearchResultsWrapper.getRowRecords(). Something is implemented improperly in the method, keep this issue open and I will take a look when I get a chance.

**Comment by xiaofan-luan** [comment]
should keep this issue and think of how to imrpvoe

**Comment by yhmo** [comment]
The performance bug of getRowRecords() has been fixed by this pr: https://github.com/milvus-io/milvus-sdk-java/pull/658
The java sdk v2.3.2 which contains this fix has been released.
The java sdk v2.2.15 also contains this fix, not yet been released, will release it along with milvus v2.2.15.
